### PR TITLE
Log Context: Add button to open the context query in a split view

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

Adds a feature to open Log Context queries in a split view.

### CODE REVIEW

File: packages/grafana-data/src/types/logs.ts [hunk 1]
```diff
@@ -131,6 +131,11 @@ export interface DataSourceWithLogsContextSupport<TQuery extends DataQuery = Dat
    */
   getLogRowContext: (row: LogRowModel, options?: LogRowContextOptions, query?: TQuery) => Promise<DataQueryResponse>;
 
+  /**
+   * Retrieve the context query object for a given log row
+   */
+  getLogRowContextQuery?: (row: LogRowModel, options?: LogRowContextOptions, query?: TQuery) => Promise<TQuery | null>;
+
```
Comment (ivanahuckova) [inline packages/grafana-data/src/types/logs.ts:138]: I'd mention that this is currently used for showing split open button in log context

File: public/app/plugins/datasource/loki/LogContextProvider.ts [hunk 2]
```diff
@@ -33,6 +33,26 @@ export class LogContextProvider {
     this.appliedContextFilters = [];
   }
 
+  getLogRowContextQuery = async (
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LogContextProvider.ts:36]: Considering this is almost identical code that we use in `getLogRowContext`, couldn't we re-use it?

File: public/app/features/explore/Logs.tsx [hunk 3]
```diff
@@ -404,14 +406,21 @@ class UnthemedLogs extends PureComponent<Props, State> {
 
     const scanText = scanRange ? `Scanning ${rangeUtil.describeTimeRange(scanRange)}` : 'Scanning...';
 
+    const contextQuery = (logsQueries ?? []).find(
+      (q) =>
+        q.refId === contextRow?.dataFrame.refId &&
+        q.datasource != null &&
+        q.datasource.type === contextRow.datasourceType
+    );
```
Comment (ivanahuckova) [inline public/app/features/explore/Logs.tsx:414]: It looks like we are using the same logic at couple of different places, so maybe we could create a util function out of this:
image

File: public/app/features/logs/components/log-context/LogRowContextModal.tsx [hunk 4]
```diff
@@ -300,6 +321,28 @@ export const LogRowContextModal: React.FunctionComponent<LogRowContextModalProps
           Showing {context.before.length} lines {logsSortOrder === LogsSortOrder.Descending ? 'after' : 'before'} match.
         </div>
       </div>
+      {getRowContextQuery && (
```
Comment (ivanahuckova) [inline public/app/features/logs/components/log-context/LogRowContextModal.tsx:324]: I have tried it with elastic and it shows the button (that does nothing when pressed) even for elasticsearch 
image

File: public/app/features/logs/components/log-context/LogRowContextModal.tsx [hunk 5]
```diff
@@ -125,13 +131,28 @@ export const LogRowContextModal: React.FunctionComponent<LogRowContextModalProps
...
 
+  const dispatch = useDispatch();
   const theme = useTheme2();
   const styles = getStyles(theme);
   const [context, setContext] = useState<{ after: LogRowModel[]; before: LogRowModel[] }>({ after: [], before: [] });
   const [limit, setLimit] = useState<number>(LoadMoreOptions[0].value!);
   const [loadingWidth, setLoadingWidth] = useState(0);
   const [loadMoreOption, setLoadMoreOption] = useState<SelectableValue<number>>(LoadMoreOptions[0]);
 
+  const getFullTimerange = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (svennergr) [reply public/app/features/logs/components/log-context/LogRowContextModal.tsx]: Fixed in [f1d6077](https://github.com/grafana/grafana/pull/66777/commits/f1d6077277ee45063b5f883553fd4d8bfd647d21)

File: public/app/features/logs/components/log-context/LogRowContextModal.tsx (hunk 4 above)
Comment (svennergr) [reply public/app/features/logs/components/log-context/LogRowContextModal.tsx]: Fixed in [e1c416a](https://github.com/grafana/grafana/pull/66777/commits/e1c416ac548d9474e6d3f020e39764b6b6338f7e)

File: public/app/features/explore/Logs.tsx (hunk 3 above)
Comment (svennergr) [reply public/app/features/explore/Logs.tsx]: Fixed in [ae74f25](https://github.com/grafana/grafana/pull/66777/commits/ae74f25435b784104fbd9283e1c53863f9ecf761)

File: public/app/plugins/datasource/loki/LogContextProvider.ts (hunk 2 above)
Comment (svennergr) [reply public/app/plugins/datasource/loki/LogContextProvider.ts]: Fixed in [e8494cd](https://github.com/grafana/grafana/pull/66777/commits/e8494cdcbb3dba269582d3f4d9dafde6dd7df796)

File: packages/grafana-data/src/types/logs.ts (hunk 1 above)
Comment (svennergr) [reply packages/grafana-data/src/types/logs.ts]: Fixed in [03b1f89](https://github.com/grafana/grafana/pull/66777/commits/03b1f89db7ac9138776090ffe0e95d4c1cc0556d)

**Review by ivanahuckova — COMMENTED** [review]
Left 1 feedback. Let me know what do you think and otherwise LGTM! 🚀

File: public/app/features/explore/LogsContainer.tsx [hunk 6]
```diff
@@ -49,28 +53,58 @@ class LogsContainer extends PureComponent<LogsContainerProps> {
     updateTimeRange({ exploreId, absoluteRange });
   };
 
-  getLogRowContext = async (row: LogRowModel, options?: any): Promise<DataQueryResponse | []> => {
+  private async callIfDefined(
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/features/explore/LogsContainer.tsx:56]: Maybe `callOptionalLogContextMethods` ?

[the remark this answers is not in the document]
Comment (svennergr) [reply public/app/features/explore/LogsContainer.tsx:56]: I am going to revert this change and live with the duplications. This also allows me to get rid of the `any`.

[the remark this answers is not in the document]
Comment (svennergr) [reply public/app/features/explore/LogsContainer.tsx]: Done in [a6d8805](https://github.com/grafana/grafana/pull/66777/commits/a6d880597abf868be0fde4697fb7c2a5bfbb2738).

**Review by ivanahuckova — APPROVED** [review]
Nice LGTM! yay for this feature!

**Pull request merged**